lib/net/ssh/config.rb

```diff
@@ -252,15 +259,20 @@ def translate_config_key(hash, key, value, settings)
             end
           when :preferredauthentications
             hash[:auth_methods] = value.split(/,/) # TODO we should place to preferred_auth_methods rather than auth_methods
-          when :proxycommand
-            if value and value !~ /^none$/
-              require 'net/ssh/proxy/command'
-              hash[:proxy] = Net::SSH::Proxy::Command.new(value)
-            end
-          when :proxyjump
-            if value
-              require 'net/ssh/proxy/jump'
-              hash[:proxy] = Net::SSH::Proxy::Jump.new(value)
+          when :proxy
+            proxy_type, value = value
+
+            case proxy_type
+            when 'proxycommand'
+              if value and value !~ /^none$/
+                require 'net/ssh/proxy/command'
+                hash[:proxy] = Net::SSH::Proxy::Command.new(value)
+              end
+            when 'proxyjump'
+              if value
```

Comment:
I think this `if value` is unneccesary we're already checking if proxy_value is truthy when we add it.

Reply:
Good point, thanks. I'll remove them.